packages/cadl-ranch-specs/http/resiliency/srv-driven/main.tsp

```diff
@@ -0,0 +1,196 @@
+import "@typespec/http";
+import "@typespec/versioning";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Versioning;
+using Azure.ClientGenerator.Core;
+using TypeSpec.Http;
+
+@versioned(Versions)
+@doc("""
+  Test that we can grow up a service spec and service deployment into a multi-versioned service with full client support.
+
+  There are three concepts that should be clarified:
+  1. Client spec version: refers to the spec that the client is generated from. 'v1' is a client generated from old.tsp and 'v2' is a client generated from main.tsp.
+  2. Service deployment version: refers to a deployment version of the service. 'v1' represents the initial deployment of the service with a single api version. 'v2' represents the new deployment of a service with multiple api versions
+  3. Api version: The initial deployment of the service only supports api version 'v1'. The new deployment of the service supports api versions 'v1' and 'v2'.
+
+  We test the following configurations from this service spec:
+  - A client generated from the second service spec can call the second deployment of a service with api version v1
+  - A client generated from the second service spec can call the second deployment of a service with api version v2
+  """)
+@client({
+  name: "ResiliencyServiceDrivenClient",
+})
+@service
+@server(
+  "http://localhost:3000/resiliency/service-driven/client:v2/service:{serviceDeploymentVersion}/api-version:{apiVersion}",
```

Comment:
I don't quite get the reason of separating the client and service version. IMO, the mock api with two api versions could already mock the service with two versions. Mock server should not be responsible for testing the behavior of the clients.

Reply:
My guess is for mockapi and dashboard to capture v1 and v2 separately (unsure).

Ideally, I would say we'd better to use api-version in query, as what Azure does, but here maybe it is more important to get mockapi to work.

Reply:
@weidongxu-microsoft for you comment on the expected behavior of this spec, the most important is the runtime test as described above. Checking there is no breaking change in the generated code between old and main is an added benefit you can also check but isn't the main purpose of this.

Reply:
I see. It seems to be useful to show the result in the dashboard.